app/models/master_file.rb

```diff
@@ -294,9 +294,11 @@ def update_progress_on_success!(encode)
     #TODO pull this from the encode
     self.date_digitized ||= Time.now.utc.iso8601
 
-    # Set duration after transcode if mediainfo fails to find.
-    # e.x. WebM files missing technical metadata
-    self.duration ||= encode.input.duration
+    # Update the duration detected by ActiveEncode
+    # because it has higher precision than mediainfo
+    # Set for the first time for files without duration
+    # e.g. WebM files missing technical metadata
+    self.duration = encode.input.duration.to_i.to_s if encode.input.duration.present?
```

Comment:
Why are we using `to_i` here? If we are looking for better precision should we not leave it as a float?

Reply:
Maybe it would make sense to add to the comment that the value should be whole number milliseconds?

Reply:
Oh, I missed/forgot that this was milliseconds and the conversion to whole seconds is later in the serialization chain. It might not hurt to update the comment that the value returned is milliseconds, but yeah we don't need greater than millisecond accuracy so `to_i` makes sense here :+1: